### ISSUE TITLE
Change wand + AI in shell = AI death

### ISSUE THREAD

**Issue by davethwave** [tgstation/tgstation#49075]
Don't quite remember what round it was, but basically it was dynamic, wiz was turning the station into xenos and stuff with a charge spell and change wand. Traitor HOP one humaned himself despite being the only human left besides the wiz. AI shell and HOP bunker in upload. Wiz breaks in and transforms shell and kills HOP. AI shells turns to human becoming a human with AI laws. Xenos kill human AI. AI dies but the actual core doesn't die yet is ghost.

So basically transform AI in shell. AI becomes x mob with AI laws. AI dies and become ghost. Core however doesn't die but dead AI can't do any AI stuff anymore since it's a ghost.

**Comment by GunHog** [comment]
Ideally, the fix for this is to somehow ensure the AI is disconnected from the shell if the mob is destroyed in any way. The gameplay intent is that the AI is controlling the shell like an RC car. Nothing that happens to the shell is supposed to happen to the AI, except being disconnected.

**Comment by TheMidnightRose** [comment]
Hopefully fixing this will also fix an issue I noticed a while back. If you die in a shell as a malf AI, even though you get yanked back to your core, your objectives consider you dead.

**Comment by zxaber** [comment]
I suspect the main issue with just ensuring the AI is disconnected is that the client is probably moved before the old mob is deleted. I guess the shell could be given free reign to just put the last client it had back into the AI core, but that sounds like a *very* dangerous idea.

Either the borg code, or else the bolt of change, will probably need a snowflake catch for when the interaction between the two happens. I'm not sure which is more proper. I suppose stuffing it into the mob code is more consistant with all the blob_act() and attack_alien() and other specialty actions that get thrown into the defense code.

AIs getting a failure on the malf objective is likely a different issue entirely.
